### ISSUE TITLE
DietPi-Backup | Enhance G_PROMPT_BACKUP

### ISSUE THREAD

**Issue by Phil1988** [MichaIng/DietPi#6796]
### Creating a bug report/issue

Required Information
DietPi version:
G_DIETPI_VERSION_CORE=8
G_DIETPI_VERSION_SUB=14
G_DIETPI_VERSION_RC=2
G_GITBRANCH='master'
G_GITOWNER='MichaIng'
G_LIVE_PATCH_STATUS[0]='not applicable'
G_LIVE_PATCH_STATUS[1]='applied'
G_LIVE_PATCH_STATUS[2]='applied'

Distro version:
bullseye

Kernel version:
Linux DietPi 5.10.0-21-amd64 # 1 SMP Debian 5.10.162-1 (2023-01-21) x86_64 GNU/Linux

SBC model:
Virtual Machine (x86_64) on Proxmox VE 7.3

#### Steps to reproduce
Tried to upgrade from bullseye to bookworm with
`sudo bash -c "$(curl -sSf 'https://raw.githubusercontent.com/MichaIng/DietPi/master/.meta/dietpi-bookworm-upgrade')"`
Unforunetly the backup failed - probably due to no space left on the disk.
After this I wanted to upgrade without backing up (its a VM and I have a snapshot as backup anyway).
Now the redis server cant be stoped.

See here:
![image](https://github.com/MichaIng/DietPi/assets/6054234/61c8075b-9729-4b59-ba68-310ed409093f)

**Improvement:**
It would be great if the backup checks if there es enough space before the start - but its obviously my mistake.

How to proceed?
If I do understand it correctly, the redis-server cant be stoped beceuase of "Write errir savin DB on disk: No space left on device".

Would be nice if you can tell me how I can delete the failed backup which was something about 120GB as it failed.
I didnt expected it fo get this big :)

Thanks in advance !

**Comment by Joulinar** [comment]
It depends on your backup location. Usually default location is within `/mnt`. Simply remove backup folder.

**Comment by Phil1988** [comment]
like this one right?
![image](https://github.com/MichaIng/DietPi/assets/6054234/08f10b69-8f1a-472e-99b2-8ba08256f8f6)

(just do be safe to not delete something important)

**Comment by Phil1988** [comment]
Thanks a lot, it works now.
Is there a simple way for you to integrate a short "availible disk space" check before doing such a the backup?
If no, then you can simply close this one as its not a really important improvement or a bug.

**Comment by MichaIng** [comment]
There was a disk space check included some DietPi versions ago. We intentionally removed it since it took much time and gave incorrect results on network drives. And we thought that an error like "no space left" is quite clear and you would know what to do?

To have disk space better in view, you can add it to the login banner via `dietpi-banner`. Or use `df -h` to get a general overview of mounts and their free/used space.

**Comment by Phil1988** [comment]
Ahh that explains it!

I remembered that "some moons ago" there was such a hint, but completely understandable that you removed it.
And yes, its obvious that files have to be deleted, but for me I didnt know what files.

Maybe it might be great that if the backup failed in this case, it automatically removes the files directly.
But surely you have more important things to do.

I will close this now and thank you for this great piece of software!

**Comment by MichaIng** [comment]
In such case it is hard to say what the admin/user wants: If there was a backup before which is just updated, then the added disk usage is usually very small, so one might want to remove other data instead, ah or try to get some space via `dietpi-cleaner` 🙂. If it is a new backup and hence incomplete, removing is more likely to be wanted, until enough space is provided/cleared otherwise. We could add some info in case of an error, or extend the error message to contain the path to the backup, with the info that it is safe to remove it from there and start over. `rsync` does not throw a particular error code when aborting due to insufficient disk space, but one which contains this case: https://manpages.debian.org/rsync#EXIT_VALUES
> 23 - Partial transfer due to error

This should be it.

> Would be nice if you can tell me how I can delete the failed backup which was something about 120GB as it failed.
I didnt expected it fo get this big :)

I read this just now. We could also add an info to the backup prompt, telling what exactly is backed up. As long as `/mnt/dietpi_userdata` is located on the rootfs, it is backed up as well, which in case contains media files, downloads and other data, which can be quite big. If userdata have been moved to an external drive, it is skipped, so then only the system itself and data in `/var/lib` and users' home dirs are backed up.

**Comment by Phil1988** [comment]
As my `/mnt/dietpi_userdata` is on the rootfs and contains like ~2TB user data, I think that was the issue here and a dietp-cleaner wouldnt make a difference :D 

So everything is fine for me and I will remember everything for future.

**Comment by MichaIng** [comment]
Makes sense then to exclude userdata with the include/exclude filters. But I think in general a simple toggle for this would be helpful. This is actually long on the ToDo already, but low-priority indeed. In this particular case, the main problem is that you likely did not use `dietpi-backup` before, hence could not know about the backup location etc, and the prompt triggered by the upgrade script (and also e.g. by `dietpi-software reinstall ...` and some other cases) does not show you the `dietpi-backup` menu with all the info, but just runs the backup with default settings when you hit "yes".

Let me open this issue as a reminder for this enhancement. We can e.g. check whether the `dietpi-backup` config file exists, and show the full menu if not, so users see the backup location, can chance it and other settings etc, before the backup runs.

**Comment by Phil1988** [comment]
Oh this would be indeed pretty useful, but understandable low priority on you side.

Its honestly unbelievable how good your support and dietpi is for such a long period!
